### PR TITLE
Throw exception when console is null and password is prompted

### PR DESCRIPTION
**Pull request by selesse**
When running `gradle uploadArchives` through the console while the Gradle daemon is running, a bizarre exception is thrown:

```
This release 1.7.1 will be signed with your GnuPG key in /home/alex/.gnupg/secring.gpg.

FAILURE: Build failed with an exception.

* What went wrong:
Cannot invoke method readPassword() on null object
```

This change modifies the code so that if a handler to a `Console` cannot be acquired, an appropriate exception is thrown.

### CODE REVIEW

File: src/main/groovy/com/bmuschko/gradle/nexus/NexusPlugin.groovy [hunk 1]
```diff
@@ -116,10 +116,13 @@ class NexusPlugin implements Plugin<Project> {
             throw new GradleException("GnuPG secret key file $keyringFile not found. Please set $SIGNING_KEYRING=/path/to/file.gpg in <USER_HOME>/.gradle/gradle.properties.")
         }
 
-        Console console = System.console()
         console.printf "\nThis release $project.version will be signed with your GnuPG key $signingKeyId in $keyringFile.\n"
+        Console console = System.console()
```
Comment (bmuschko) [inline src/main/groovy/com/bmuschko/gradle/nexus/NexusPlugin.groovy]: You are trying to use `console.printf` before the instance exists.

**Comment by bmuschko** [comment]
Can you change your pull request to use the inner class `ConsoleHandler`?

**Comment by selesse** [comment]
I've started changing the code to use `ConsoleHandler`, but I don't think this will address the problem.

The user won't be prompted to enter their GnuPG passphrase if the `Console` is null (i.e. when the Gradle daemon is enabled), and `ConsoleHandler.askForPassword` returns null when the `Console` is null.

Empty GnuPG passphrases are discouraged, but possible, so I'm not exactly sure how to solve this problem. As a user, I'd like the plugin to warn me or prevent me from signing without providing my passphrase key, even when running the Gradle daemon.

 Any thoughts or ideas?

**Comment by selesse** [comment]
:+1: 

Let me know if you'd like me to rebase. Thanks! :)

**Comment by Johnlon** [comment]
Yes - was confused seeing the same message myself today.

**Comment by bmuschko** [comment]
I think we should completely extract the handling of credentials from this plugin. There should be a credentials plugin that does all the work here. The only thing that should be handled right now is the case where username or password might be null due to the use of the Gradle daemon.

**Comment by selesse** [comment]
Does this seem like an intermediary solution until such a plugin can be produced/discovered?

**Pull request closed without merging**